### PR TITLE
Make current Version pluggable for serverless

### PR DESCRIPTION
**Pull request by rjernst**
Version.CURRENT is statically loaded as a constant early during startup. Yet serverless needs to be able to override the current Version so it can add additional versions. This commit makes Version.CURRENT pluggable via SPI. Note that the only way for this to be plugged in is via an additional jar on the boot layer.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/TransportVersion.java [hunk 1]
```diff
@@ -263,6 +264,15 @@ public static TransportVersion fromString(String str) {
         return TransportVersion.fromId(Integer.parseInt(str));
     }
 
+    // finds the pluggable current version, or falls back to the locally defined current
+    private static TransportVersion findCurrent(TransportVersion fallback) {
+        var versionExtension = VersionExtension.load();
+        if (versionExtension == null) {
+            return fallback;
+        }
+        return new TransportVersion(versionExtension.getCurrentTransportVersionId());
```
Comment (ChrisHegarty) [inline server/src/main/java/org/elasticsearch/TransportVersion.java:273]: The comment for this method is not quite right? the fallback is to the passed in fallback, which I guess will be the CURRENT.

**Review by ChrisHegarty — APPROVED** [review]
LGTM. Just left one small comment.

File: server/src/main/java/org/elasticsearch/TransportVersion.java [hunk 2]
```diff
@@ -263,6 +264,15 @@ public static TransportVersion fromString(String str) {
         return TransportVersion.fromId(Integer.parseInt(str));
     }
 
+    // finds the pluggable current version, or falls back to the locally defined current
+    private static TransportVersion findCurrent(TransportVersion fallback) {
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/TransportVersion.java:268]: This isn't called from anywhere yet?

Comment (rjernst) [reply server/src/main/java/org/elasticsearch/TransportVersion.java:268]: Oops, fixed.

File: server/src/main/java/org/elasticsearch/TransportVersion.java (hunk 1 above)
Comment (rjernst) [reply server/src/main/java/org/elasticsearch/TransportVersion.java:273]: I reworded the comment.

**Review by thecoop — APPROVED** [review]

**Pull request merged**